src/lib/storage/index/table_index_type.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <boost/hana/at_key.hpp>
+
+// #include "all_type_variant.hpp"
+
+namespace hyrise {
+
+namespace hana = boost::hana;
+
+enum class TableIndexType : uint8_t { PartialHash };
+
+class PartialHashIndex;
+
+namespace detail {
+
+constexpr auto table_index_map =
+    hana::make_map(hana::make_pair(hana::type_c<PartialHashIndex>, TableIndexType::PartialHash));
+
+}  // namespace detail
+
+template <typename IndexType>
```

Comment:
Great idea!
AbstractTableIndex already held its type.
I've removed the template now.
Because of this, we also were able to get rid of the explicit instantiation in table.cpp :)